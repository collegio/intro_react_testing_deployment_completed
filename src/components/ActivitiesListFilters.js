import React from 'react';
import { connect } from 'react-redux';
import { setFilterText, setFilterType, sortByName, sortByDistance } from '../actions/filters';

export class ActivitiesListFilters extends React.Component {

    constructor(props) {
        super(props);

        this.onTextChange = this.onTextChange.bind(this);
        this.onSortChange = this.onSortChange.bind(this);
        this.onActivityTypeChange = this.onActivityTypeChange.bind(this);
    }

    onTextChange(e) {
        this.props.setFilterText(e.target.value);
    }

    onSortChange(e) {
        if (e.target.value === "name") {
            this.props.sortByName();
        }
        else if (e.target.value === "distance") {
            this.props.sortByDistance();
        }
    }

    onActivityTypeChange(e) {
        this.props.setFilterType(e.target.value);
    }

    render() {
        return (
            <div className="activities-list-filters">
                <h2>Filter Activities</h2>
                <div className="filter-group">
                    <label>Search For:</label><br />
                    <input type="text" value={this.props.filters.text} onChange={this.onTextChange} />
                </div>
                <div className="filter-group">
                    <label>Sort By:</label><br />
                    <select onChange={this.onSortChange}>
                        <option value="name">Name</option>
                        <option value="distance">Distance</option>
                    </select>
                </div>
                <div className="filter-group">
                    <label>Sport Type:</label><br />
                    <select onChange={this.onActivityTypeChange}>
                        <option value="all">All</option>
                        <option value="running">Running</option>
                        <option value="walking">Walking</option>
                        <option value="cycling">Cycling</option>
                        <option value="swimming">Swimming</option>
                    </select>
                </div>
            </div>
        );
    }
}

const mapStateToProps = (state) => {
    return {
        filters: state.filters
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        setFilterText: (text) => dispatch(setFilterText(text)),
        setFilterType: (activity_type) => dispatch(setFilterType(activity_type)),
        sortByName: () => dispatch(sortByName()),
        sortByDistance: () => dispatch(sortByDistance()),
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(ActivitiesListFilters);