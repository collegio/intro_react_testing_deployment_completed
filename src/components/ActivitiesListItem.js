import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { removeActivity } from '../actions/activities';

export const ActivitiesListItem = ({ dispatch, id, name, activity_type, distance }) => (
    <div className="activity">
        <div className="activity-info">
            <p>
                <Link to={'/edit/'+id}>
                    {name}<br />
                </Link>
                {activity_type} - {distance}
            </p>
            <button className="button button-clear" onClick={() => {
                dispatch(removeActivity({ id }));
            }}>Remove</button>
        </div>
    </div>
);

export default connect()(ActivitiesListItem);