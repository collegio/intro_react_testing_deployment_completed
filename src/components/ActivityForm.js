import React from 'react';

export class ActivityForm extends React.Component {

    constructor(props) {
        super(props);

        this.onNameChange = this.onNameChange.bind(this);
        this.onActivityChange = this.onActivityChange.bind(this);
        this.onDistanceChange = this.onDistanceChange.bind(this);
        this.onSubmitActivity = this.onSubmitActivity.bind(this);

        this.state = {
            name: props.activity ? props.activity.name : '',
            activity_type: props.activity ? props.activity.activity_type : 'walking',
            distance: props.activity ? props.activity.distance : '0 km',
            error: ''
        }
    }

    onNameChange(e) {
        const name = e.target.value;
        this.setState(() => ({ name }));
    };

    onActivityChange(e) {
        const activity_type = e.target.value;
        this.setState(() => ({ activity_type }));
    };

    onDistanceChange(e) {
        const distance = e.target.value;
        this.setState(() => ({ distance }));
    };

    onSubmitActivity(e) {
        e.preventDefault();

        if (!this.state.name) {

            // Error!
            this.setState(() => ({ error: 'Please enter a name first.' }));
        }
        else {

            // Remove the error and Submit!
            this.setState(() => ({ error: '' }));
            this.props.onSubmit({
                name: this.state.name,
                activity_type: this.state.activity_type,
                distance: this.state.distance
            });
        }
    }

    render() {
        return (
            <div className="activity-form">
                { this.state.error && <p>{this.state.error}</p> }
                <form onSubmit={this.onSubmitActivity}>
                    <label>Name</label><br/>
                    <input type="text" placeholder="Name" value={this.state.name} onChange={this.onNameChange} autoFocus />

                    <label>Sport</label><br/>
                    <select value={this.state.sport_type} onChange={this.onActivityChange}>
                        <option value="running">Running</option>
                        <option value="walking">Walking</option>
                        <option value="cycling">Cycling</option>
                        <option value="swimming">Swimming</option>
                    </select>

                    <label>Distance</label><br/>
                    <input type="text" placeholder="Name" value={this.state.distance} onChange={this.onDistanceChange} autoFocus />

                    <button className="button pull-right">Submit</button>
                </form>
            </div>
        );
    }
}

export default ActivityForm;