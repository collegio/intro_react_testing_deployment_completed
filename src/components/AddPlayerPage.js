import React from 'react';
import { connect } from 'react-redux';
import PlayerForm from './PlayerForm';
import { addPlayer } from '../actions/players';

export class AddPlayerPage extends React.Component {

    constructor(props) {
        super(props);

        this.onSubmit = this.onSubmit.bind(this);
    }

    onSubmit(player) {
        this.props.addPlayer(player);
        this.props.history.push('/');
    }

    render() {
        return (
            <div className="container">
                <h1>Add a Player</h1>
                <PlayerForm 
                    onSubmit={this.onSubmit}
                />
            </div>
        );
    }
}

const mapDispatchToProps = (dispatch) => ({
    addPlayer: (player) => dispatch(addPlayer(player))
});

export default connect(undefined, mapDispatchToProps)(AddPlayerPage);