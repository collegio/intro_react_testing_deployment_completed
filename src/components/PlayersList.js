import React from 'react';
import { connect } from 'react-redux';
import PlayersListItem from './PlayersListItem';
import selectPlayers from '../selectors/players';

export const PlayersList = (props) => (
    <div className="players-list">
        <h1>Players List</h1>
        { props.players.length === 0 ? (
            <p>No players available!</p>
        ) : (
            <div className="player-list">
                {props.players.map((player) => {
                    return <PlayersListItem key={player.id} {...player} />;
                })}
            </div>
        )}
    </div>
);

const mapStateToProps = (state) => {
    return {
        players: selectPlayers(state.players, state.filters)
    }
};

export default connect(mapStateToProps)(PlayersList);