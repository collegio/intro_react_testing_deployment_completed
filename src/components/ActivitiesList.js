import React from 'react';
import { connect } from 'react-redux';
import ActivitiesListItem from './ActivitiesListItem';
import selectActivities from '../selectors/activities';

export const ActivitiesList = (props) => (
    <div className="activities-list">
        <h1>Activities List</h1>
        <div className="activity-list">
            {props.activities.map((activity) => {
                return <ActivitiesListItem key={activity.id} {...activity} />;
            })}
        </div>
    </div>
);

const mapStateToProps = (state) => {
    return {
        activities: selectActivities(state.activities, state.filters)
    }
};

export default connect(mapStateToProps)(ActivitiesList);