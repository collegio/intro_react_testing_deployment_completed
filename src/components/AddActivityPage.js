import React from 'react';
import { connect } from 'react-redux';
import ActivityForm from './ActivityForm';
import { addActivity } from '../actions/activities';

export class AddActivityPage extends React.Component {

    constructor(props) {
        super(props);

        this.onSubmit = this.onSubmit.bind(this);
    }

    onSubmit(activity) {
        this.props.addActivity(activity);
        this.props.history.push('/');
    }

    render() {
        return (
            <div className="container">
                <h1>Add a Activity</h1>
                <ActivityForm 
                    onSubmit={this.onSubmit}
                />
            </div>
        );
    }
}

const mapDispatchToProps = (dispatch) => ({
    addActivity: (activity) => dispatch(addActivity(activity))
});

export default connect(undefined, mapDispatchToProps)(AddActivityPage);