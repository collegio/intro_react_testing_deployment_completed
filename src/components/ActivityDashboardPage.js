import React from 'react';
import ActivitiesList from './ActivitiesList';
import ActivitiesListFilters from './ActivitiesListFilters';

export const ActivityDashboardPage = (props) => (
    <div className="container">
        <ActivitiesListFilters />
        <ActivitiesList />
    </div>
);

export default ActivityDashboardPage;