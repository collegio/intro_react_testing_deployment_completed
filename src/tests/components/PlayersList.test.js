import React from 'react';
import { shallow } from 'enzyme';
import { PlayersList } from '../../components/PlayersList';
import players from '../fixtures/players';

test('should render the players list with players', () => {
    const wrapper = shallow(<PlayersList players={players} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render the players list no players', () => {
    const wrapper = shallow(<PlayersList players={[]} />);
    expect(wrapper).toMatchSnapshot();
});