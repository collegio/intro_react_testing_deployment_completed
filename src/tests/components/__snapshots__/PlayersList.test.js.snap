// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the players list no players 1`] = `
<div
  className="players-list"
>
  <h1>
    Players List
  </h1>
  <p>
    No players available!
  </p>
</div>
`;

exports[`should render the players list with players 1`] = `
<div
  className="players-list"
>
  <h1>
    Players List
  </h1>
  <div
    className="player-list"
  >
    <Connect(PlayersListItem)
      gender="Male"
      id="1"
      key="1"
      message="Just looking to play"
      name="Sidney Crosby"
      skill_level="expert"
      sport_type="hockey"
    />
    <Connect(PlayersListItem)
      gender="Male"
      id="2"
      key="2"
      message="Test test test"
      name="Lebron James"
      skill_level="expert"
      sport_type="basketball"
    />
    <Connect(PlayersListItem)
      gender="Male"
      id="3"
      key="3"
      message="test"
      name="Drew Brees"
      skill_level="expert"
      sport_type="football"
    />
    <Connect(PlayersListItem)
      gender="Female"
      id="4"
      key="4"
      message=""
      name="Hayley Wickenheiser"
      skill_level="expert"
      sport_type="hockey"
    />
    <Connect(PlayersListItem)
      gender="Male"
      id="5"
      key="5"
      message="I'm terrible!"
      name="Johnny Brees"
      skill_level="basic"
      sport_type="basketball"
    />
  </div>
</div>
`;
